budgetApp1/budgetApp1/CoreDataHandler.swift

```diff
@@ -0,0 +1,73 @@
+//
+//  CoreDataHandler.swift
+//  budgetApp1
+//
+//  Created by Lehlohonolo Nkadimeng on 2024/05/02.
+//
+
+import CoreData
+import UIKit
+
+protocol CoreDataHandlerType {
+    func createUser(fullName: String, password: String, emailAddress: String, username: String) -> Bool
+    func checkIfUserHasAccount(username: String, password: String) -> Bool
+}
+
+class CoreDataHandler: CoreDataHandlerType {
+    
+    weak var appDelegate: AppDelegate?
+    let context: NSManagedObjectContext
+    
+    init() {
+        guard let appDelegate = UIApplication.shared.delegate as? AppDelegate else {
+            fatalError("Unable to access AppDelegate")
+        }
+        self.appDelegate = appDelegate
+        context = appDelegate.persistentContainer.viewContext
+    }
+    
+    // MARK: - Public CRUD functions
+    
+    func checkIfUserHasAccount(username: String, password: String) -> Bool {
+        
+        let fetchRequest: NSFetchRequest<UserEntity> = UserEntity.fetchRequest()
+        fetchRequest.predicate = NSPredicate(format: "username == %@ AND password == %@", username, password)
+        do {
+            let results = try context.fetch(fetchRequest)
+            if !results.isEmpty {
+                return true
+            } else {
+                return false
+            }
```

Comment:
Please again don't use an if with a bool `!result.isEmpty` to return true and false

You can just `return !result.isEmpty`